### ISSUE TITLE
Respect page models

### ISSUE THREAD

**Issue by lukasbestle** [getkirby-plugins/algolia-plugin#3]
See #2.

**Comment by pedroborges** [comment]
@lukasbestle I found a way to implement this, but it requires a non-breaking addition to the Kirby's core. We first need a `model` method on the Page object:

```php
// kirby/core/page.php

  /**
   * Creates a page model
   */
  public function model() {

    if(!isset(static::$models[$this->template()])) {
      kirby()->models();
    }

    $model = static::$models[$this->template()];
    $page = new $model($this->parent(), $this->dirname());

    return $page;

  }
```

Then we need this one line added to the `formatPage` method on the plugin:

```php
public static function formatPage(Page $page) {
    $page = $page->model();

    // ...
```

This is not the first time I have issues with page models not working inside hooks, the addition the `model` method be very useful for plugin developers. If this sounds good I would gladly submit a PR to the core.

**Comment by lukasbestle** [comment]
Sounds good. You don't need to send a PR as you already posted the code here. :)

**Comment by pedroborges** [comment]
Hadn't thought of using page method, it will do the trick while we wait for 2.4. Thanks 😄

**Comment by jpangborn** [comment]
Should this issue be closed? I am using Kirby 2.4.1 and this seems to be working.

**Comment by lukasbestle** [comment]
@pedroborges Can you confirm this?
